diff --git a/cpp/memilio/io/result_io.cpp b/cpp/memilio/io/result_io.cpp
index 9fc3aa23b..a0612ab90 100644
--- a/cpp/memilio/io/result_io.cpp
+++ b/cpp/memilio/io/result_io.cpp
@@ -108,6 +108,11 @@ IOResult<std::vector<SimulationResult>> read_result(const std::string& filename)
     MEMILIO_H5_CHECK(H5Literate(file.id, H5_INDEX_NAME, H5_ITER_INC, NULL, &store_group_name, &h5group_names),
                      StatusCode::UnknownError, "Group names could not be read.");
 
+    std::sort(h5group_names.begin(), h5group_names.end(),
+            [](const std::string &a, const std::string &b) {
+                return std::stof(a) < std::stof(b);
+            });
+
     for (auto& h5group_name : h5group_names) {
         H5Group region_h5group{H5Gopen(file.id, h5group_name.c_str(), H5P_DEFAULT)};
         MEMILIO_H5_CHECK(region_h5group.id, StatusCode::UnknownError,
diff --git a/cpp/memilio/mobility/graph.h b/cpp/memilio/mobility/graph.h
index fade2f3a4..a7a60330e 100644
--- a/cpp/memilio/mobility/graph.h
+++ b/cpp/memilio/mobility/graph.h
@@ -426,7 +426,7 @@ IOResult<void> set_edges(const fs::path& mobility_data_file, Graph<Model, Mobili
             commuting_weights =
                 (commuting_weights.size() == 0 ? std::vector<FP>(num_age_groups, 1.0) : commuting_weights);
             //commuters
-            auto working_population = 0.0;
+            FP working_population = 0.0;
             for (auto age = AgeGroup(0); age < populations.template size<mio::AgeGroup>(); ++age) {
                 working_population += populations.get_group_total(age) * commuting_weights[size_t(age)];
             }
